TUT/STA130F24_TUT02_Sep13.ipynb

```diff
@@ -2,101 +2,72 @@
-    "4. Tell ChatGPT that you have a couple extensions of the code to show it, and show it each of these of the algorithm in sequence\n",
-    "    1. Without giving it the answers, see if you can prompt ChatGPT to figure out the extensions in the code; namely, the trigram (rather than bigram) dependency, and the character specific Markov chains \n",
-    "    2. See if ChatGPT can suggest any streamlining or readability or usability improvments to the code\n",
-    "    3. See what ChatGPT is able to help you understand about the code and what's hard to understand even when  ChatGPTs is helping to explaining things\n",
-    "    4. Start a new ChatGPT and first show it the original Markov Chatbot, and then tell ChatGPT that you have an extension but just show it Extension #2 directly without showing it Extension #2 and see if it's still able to understand everything extension does. \n",
-    "5. Report on your experience with ChatGPT discussing how quickly it was able to be helpful for each of the above questions, and if so, how\n",
-    "6. Reflect on your experience interacting with ChatGPT and describe how your perception of AI-driven assistance tools in the context of learning coding, statistics, and data science may be evolving\n",
-    "7. By searching online and discussing the topic with ChatGPT, describe the relevance of learning and adaptability, communication, coding, and statistics and data analysis as skills in the modern world, epseically with respect to career opportunities (particularly in the context of the data science industry)\n",
-    "8. See if ChatGPT thinks you could be a statistician or a data scientist without coding or doing data analysis and summarise the assessment ChatGPT provides\n",
-    "9. Do some career exploration with ChatGPT to identify the skills that might be the most valuable for a career that you're interested in and paraphrase the conclusions of your conversation"
-   ]
+    "1. `for i in range(n)` , `for x in enumerate(lst)` and  `for i,x in enumerate(lst)`\n",
+    "2. `list()`, `dict()` and `tuple()`\n",
+    "3. NumPy functions:\n",
+    "    1. `import numpy as np`\n",
+    "    2. `np.array()`\n",
+    "    3. `np.random.choice([1,2,3])`\n",
+    "4. `if`/`else` conditional statements"
+   ],
+   "metadata": {
+    "collapsed": false
+   },
+   "id": "7b6f2d524b9e6c7e"
+  },
+  {
+   "cell_type": "code",
+   "execution_count": null,
+   "outputs": [],
+   "source": [
+    "import numpy as np\n",
+    "np.random.choice([1,2,3])"
+   ],
+   "metadata": {
+    "collapsed": false
+   },
+   "id": "20e8c519ed45cbfb"
   },
   {
    "cell_type": "code",
-   "execution_count": 13,
-   "id": "f8f78323",
-   "metadata": {},
+   "execution_count": 1,
    "outputs": [
     {
      "data": {
-      "text/plain": [
-       "3"
-      ]
+      "text/plain": "0.66777"
```

Comment:
hmm weird, seems out of place? There's not code for it? I'll clean it up later if needed.